test/functional/clusteradm/clusteradmE2E_test.go

```diff
@@ -0,0 +1,266 @@
+// Copyright Contributors to the Open Cluster Management project
+package clusteradme2e
+
+import (
+	"testing"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	"open-cluster-management.io/clusteradm/test/functional/util"
+)
+
+func TestE2EClusteradm(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t, "E2E clusteradm test")
+}
+
+var _ = ginkgo.Describe("test clusteradm with service account", func() {
+	defer ginkgo.GinkgoRecover()
+
+	var e2e *util.TestE2eConfig
+	ginkgo.By("With Service Account")
+
+	ginkgo.BeforeEach(func(done ginkgo.Done) {
+		ginkgo.By("bootstrapping e2e test environment")
+		// get cluster infos
+		e2e = util.BootStrapE2eEnv("sa", true)
+		close(done)
+	}, 1200)
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By("tearing down e2e test environment")
+		// clean
+		e2e.ClearE2eEnv()
+
+	})
+
+	ginkgo.Context("join hub scenario", func() {
+		ginkgo.By("join hub with service account")
+
+		ginkgo.It("should managedcluster join hub successfully", func() {
+			ginkgo.By("clusteradm version check")
+			err := e2e.Clusteradm().Version().Run()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "clusteradm version check error")
+
+			ginkgo.By("init hub")
+			jn, err := e2e.Clusteradm().Init(
+				"--context", e2e.Cluster().Hub().Context(),
```

Comment:
the kind clusters are created before run e2e, right? 
could we set the info of clusters as a variables, make it easy to understand, or add more comments?